internal/commands/build/types.go

```diff
@@ -0,0 +1,73 @@
+package build
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/docker/cli/cli/compose/loader"
+	compose "github.com/docker/cli/cli/compose/types"
+	"github.com/pkg/errors"
+)
+
+// A minimal subset of github.com/docker/cli/cli/compose/types/types.go for the purpose of loading the build configuration
+
+type ServiceConfig struct {
+	Name string `yaml:"-" json:"-"`
+
+	Build *ImageBuildConfig
+	Image *string
+}
+
+type ImageBuildConfig struct {
+	Context    string                    `yaml:",omitempty" json:"context,omitempty"`
+	Dockerfile string                    `yaml:",omitempty" json:"dockerfile,omitempty"`
+	Args       compose.MappingWithEquals `yaml:",omitempty" json:"args,omitempty"`
+}
+
+func Load(dict map[string]interface{}) ([]ServiceConfig, error) {
+	section, ok := dict["services"]
+	if !ok {
```

Comment:
nit: I think you can remove this check, as the cast on `section` under is valid on `nil` value, it won't panic.

Reply:
but then we will pas `nil` further to `loader.Transform` which I can't tell will be fine. Better safe than sorry